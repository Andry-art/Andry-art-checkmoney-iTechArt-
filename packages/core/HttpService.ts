/* eslint-disable consistent-return */
/* eslint-disable no-unused-vars */
/* eslint-disable max-classes-per-file */


type TokensData = {
  accessToken: string;
  refreshToken: string;
  ok?:boolean;
}

// eslint-disable-next-line no-unused-vars
type FetchCallback = (url: string, param?: object) => Promise<Response>;
type SaveTokenCallback = (response: Promise<any>) => TokensData;
type GetTokenCallback = () => TokensData;

export class ErrorFetch extends Error {
  network: string | number;
  constructor(message: string, network: string | number) {
    super(message);
    this.message = message;
    this.network = network;
  }
}

class HttpService {
  private fetch: FetchCallback;
  private FetchResponse: FetchCallback;

  private getToken: GetTokenCallback;

  private saveToken: SaveTokenCallback;

  constructor(FetchCallbackResponseFun: FetchCallback, fetchFunc: FetchCallback, getTokensFunc: GetTokenCallback, saveTokensFunc: SaveTokenCallback) {
    this.FetchResponse = FetchCallbackResponseFun,
    this.fetch = fetchFunc;
    this.getToken = getTokensFunc;
    this.saveToken = saveTokensFunc;
  }

 

  public async authGet(url: string, limit: number = 1): Promise<object|undefined> {
    const tokens = this.getToken();
    let response 

    try {
      response = await this.fetch(url, {
        method: 'GET',
        headers: {
          Authorization: `Bearer ${tokens.accessToken}`,
        },
      });

      if (response.ok) {
        const responseJSON = await response.json();
        return responseJSON
      } 
       if (response.status === 401 && limit === 1) {
        await this.refreshToken(tokens.refreshToken);
        return await this.authGet(url, 0);
      } 
       if(response.status === 401){
        throw new ErrorFetch('authGet error1', response.status);
      }
      throw new ErrorFetch('authGet error2', 'error');
      
    } catch (error) {
    
         throw error
    }
  }


  public async authPost<T>(url: string, params: any, limit: number = 1): Promise<T | undefined> {
    const tokens = this.getToken();
    let response;

    try {
      response = await this.fetch(url, {
        method: 'POST',
        headers: {
          Authorization: `Bearer ${tokens.accessToken}`,
        },
        body: JSON.stringify({
          params,
        }),
      });

      if (response.ok) {
        const responseJSON = await response.json();
        return responseJSON
      }

      if (response.status === 401 && limit === 1) {
        await this.refreshToken(tokens.refreshToken);
       return await this.authPost(url, params, limit - 1);
      }
      if(response.status === 401){
        throw new ErrorFetch('authPost error1', response.status);
      }
      throw new ErrorFetch('authPost error2', ' error');
    } catch (error) {
    
      throw error
    }
  }

  public async authPut<T>(url: string, params: any, limit: number = 1): Promise<T | undefined> {
    const tokens = this.getToken();
    let response;

    try {
      response = await this.fetch(url, {
        method: 'PUT',
        headers: {
          Authorization: `Bearer ${tokens.accessToken}`,
        },
        body: JSON.stringify({
          params,
        }),
      });

      if (response.ok) {
        const responseJSON = await response.json();
        return responseJSON
      }

      if (response.status === 401 && limit === 1) {
        await this.refreshToken(tokens.refreshToken);
        await this.authPut(url, params, limit - 1);
      } if(response.status === 401){
        throw new ErrorFetch('authPut error1', response.status);
      }

      throw new ErrorFetch('authPut error2', 'network error');
    } catch (error) {
     
      throw error
        
    }
  }

  public async authDelete(url: string, params: any, limit: number = 1): Promise<void> {
    const tokens = this.getToken();
    let response;

    try {
      response = await this.fetch(url, {
        method: 'DELETE',
        headers: {
          Authorization: `Bearer ${tokens.accessToken}`,
        },
        body: JSON.stringify({
          params,
        }),
      });

      if(response.status === 200){
        return await response.json()
      }

      if (response.status === 401 && limit === 1) {
        await this.refreshToken(tokens.refreshToken);
        await this.authDelete(url, params, limit - 1);
      }
      if (response.status === 401) {
        throw new ErrorFetch('authDelete error1', response.status);
      }

      throw new ErrorFetch('authDelete error2', 'network error');
    } catch (error) {
     
        throw error
        
    }
  }

  public async auth<T>(url: string, email: string, password: string): Promise<T | undefined> {
    let response;

    try {
      response = await this.fetch(url, {
        method: 'POST',
        headers: {
          Accept: 'application/json',
          'Content-Type': 'application/json',
        },
        body: JSON.stringify({
          email,
          password,
        }),
      });

      if (response.ok) {
       const responseData = await response.json();
        this.saveToken(responseData);
        return responseData
      }
      if(response.status === 401){
      throw new ErrorFetch('auth error1', response.status);
      }

      throw new ErrorFetch('auth error2', 'network error');
    } catch (error) {
      
        throw error
    }
  }

  public async refreshToken(token: string): Promise<any> {
    let response;

    try {
      response = await this.FetchResponse('api/auth/refreshToken', {
        method: 'POST',
        headers: {
          Accept: 'application/json',
          'Content-Type': 'application/json',
        },
        body: JSON.stringify({
          token,
        }),
      })

      if (response.ok) {
       let responseJSON = await response.json()
        this.saveToken(responseJSON);
       return responseJSON
      } 
      if(response.status === 201){
      throw new ErrorFetch('refreshToken error1', response.status);
      }
     
      throw new ErrorFetch('refreshToken error2', 'network error');
    } catch (error) {
      console.log(error)
        throw error
    }
  }
}

export default HttpService;
