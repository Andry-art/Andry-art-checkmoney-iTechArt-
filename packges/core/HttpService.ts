

type FetchCallback = (url: string, param?: object) => Promise<Response>;


class ErrorFetch extends ErrorEvent {
    constructor({ message, stack, colno, error, lineno, filename }:
        { message: string; stack?: string; colno?: number; error?: string, lineno?: string, filename?: string }) {
        super(message)
        message
        colno
        stack
        error
        lineno
        filename
    }
}



let tokenData: Promise<any>;

export function saveToken(response: Promise<any>): Promise<void> {
    return tokenData = response;
}

export const getToken = (): object => {
    return tokenData
}



class HttpService {

    private fetch: FetchCallback
    private getToken: Function
    private saveToken: Function

    constructor(fetchFync: FetchCallback, getToken: Function, saveToken: Function) {
        this.fetch = fetchFync
        this.getToken = getToken;
        this.saveToken = saveToken;
    }

    public async authGet(url: string, limit: number = 1): Promise<any> {
        let token = this.getToken()
        let tokenAccess = token.accessToken

        let response = await this.fetch(url, {
            method: 'GET',
            credentials: 'include',
            headers: {
                Authorization: `Bearer ${tokenAccess}`
            }
        })

        if (response.ok) {
            return await response.json()

        } if (response.status === 401 && limit == 1) {
            await this.refreshToken(token.refreshToken)
            await this.authGet(url, limit - 1)

        } else {
            new ErrorFetch({ message: 'didn`t get request' })
        }
    }

    public async authPost(url: string, params: any, limit: number = 1): Promise<void> {
        let token = this.getToken()
        let tokenAccess = token.accessToken

        const response = await this.fetch(url, {
            method: 'POST',
            credentials: 'include',
            headers: {
                Authorization: `Bearer ${tokenAccess}`
            },
            body: JSON.stringify({
                params
            }),
        })

        if (response.ok) {
            return await response.json()
        }

        if (response.status === 401 && limit == 1) {
            await this.refreshToken(token.refreshToken)
            await this.authPost(url, limit - 1)

        } else {
            new ErrorFetch({ message: 'didn`t get request' })

        }
    }

    public async authPut(url: string, params: any, limit: number = 1): Promise<any> {
        let token = this.getToken()
        let tokenAccess = token.accessToken

        const response = await this.fetch(url, {
            method: 'PUT',
            credentials: 'include',
            headers: {
                Authorization: `Bearer ${tokenAccess}`
            },
            body: JSON.stringify({
                params
            }),
        })

        if (response.ok) {
            return await response.json()
        }

        if (response.status === 401 && limit == 1) {
            await this.refreshToken(token.refreshToken)
            await this.authPost(url, limit - 1)

        } else {
            new ErrorFetch({ message: 'didn`t get request' })
        }
    }

    public async authDelete(url: string, params: any, limit: number = 1): Promise<any> {
        let token = this.getToken()
        let tokenAccess = token.accessToken

        const response = await this.fetch(url, {
            method: 'DELETE',
            credentials: 'include',
            headers: {
                Authorization: `Bearer ${tokenAccess}`
            },
            body: JSON.stringify({
                params
            }),
        })

        if (response.status === 401 && limit == 1) {
            await this.refreshToken(token.refreshToken)
            await this.authDelete(url, limit - 1)

        }
        if (!response.ok) {
            new ErrorFetch({ message: 'didn`t get request' })
        }
    }

    public async auth(url: string, email: string, password: string): Promise<any> {
        let response = await this.fetch(url, {
            method: 'POST',
            credentials: 'include',
            headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json',
            },
            body: JSON.stringify({
                email,
                password,
            }),
        })

        if (response.ok) {
            await this.saveToken(response.json())
            return await response.json()

        } else {
            new ErrorFetch({ message: 'didn`t get request' })
        }
    }


    public async refreshToken(token: string): Promise<any> {
        let response = await this.fetch('api/auth/refreshToken', {
            method: 'POST',
            credentials: 'include',
            headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json',
            },
            body: JSON.stringify({
                token
            }),
        })

        if (response.ok) {
            await this.saveToken(response.json())
            return await response.json()

        } else {
            new ErrorFetch({ message: 'didn`t get request' })
        }
    }
}

export default HttpService;

