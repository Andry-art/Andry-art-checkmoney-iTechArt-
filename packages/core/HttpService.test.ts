
/* eslint-disable no-shadow */
/* eslint-disable no-undef */
/* eslint-disable max-len */



import HttpService from './HttpService';
import {ErrorFetch} from './HttpService';



  it('authGet fetch', async () => {
    const mockFetch = jest.fn().mockReturnValue({ok: true, json: () => {}});
    const saveToken = jest.fn().mockReturnValue({});
    const getToken = jest.fn().mockReturnValue({accessToken: 'wvwvw', refreshToken: "dfbdfbfdfb"});
    const http = new HttpService(mockFetch, getToken, saveToken);
    const tokens = getToken();

    await http.authGet('https://jsonplaceholder.typicode.com/posts');
    
    expect(mockFetch).toBeCalledWith('https://jsonplaceholder.typicode.com/posts', {
      method: 'GET',
      headers: { Authorization: `Bearer ${tokens.accessToken}` },
    });
  })


  it('authPost fetch', async () => {
    const mockFetch = jest.fn().mockReturnValue({ok: true, json: () => {}});
    const saveToken = jest.fn().mockReturnValue({});
    const getToken = jest.fn().mockReturnValue({accessToken: 'wvwvw', refreshToken: "dfbdfbfdfb"});
    const http = new HttpService(mockFetch, getToken, saveToken);
    const tokens = getToken();
    const params = { 'Accept': 'application/json', 'Content-Type': 'application/json' }

    await http.authPost('https://jsonplaceholder.typicode.com/posts', params)
    expect(mockFetch).toBeCalledWith('https://jsonplaceholder.typicode.com/posts', {
              method: 'POST',
              headers: { Authorization: `Bearer ${tokens.accessToken}`},
              body: JSON.stringify(params),
          });
  })

  it('authPut fetch', async () => {
    const mockFetch = jest.fn().mockReturnValue({ok: true, json: () => {}});
    const saveToken = jest.fn().mockReturnValue({ json: () => { } });
    const getToken = jest.fn().mockReturnValue({accessToken: 'wvwvw', refreshToken: "dfbdfbfdfb"});
    const http = new HttpService(mockFetch, getToken, saveToken);
    const tokens = getToken();
    const params = { 'Accept': 'application/json', 'Content-Type': 'application/json' }

    await http.authPut('https://jsonplaceholder.typicode.com/posts', params)
    expect(mockFetch).toBeCalledWith('https://jsonplaceholder.typicode.com/posts', {
              method: 'PUT',
              headers: { Authorization: `Bearer ${tokens.accessToken}`},
              body: JSON.stringify(params),
          });
  })


  it('authDelete fetch', async () => {
    const mockFetch = jest.fn().mockReturnValue({ok: true, status: 200, json: () => {}});
    const saveToken = jest.fn().mockReturnValue({ json: () => { } });
    const getToken = jest.fn().mockReturnValue({accessToken: 'wvwvw', refreshToken: "dfbdfbfdfb"});
    const http = new HttpService(mockFetch, getToken, saveToken);
    const tokens = getToken();
    const params = { 'Accept': 'application/json', 'Content-Type': 'application/json' }

    await http.authDelete('https://jsonplaceholder.typicode.com/posts', params)
    expect(mockFetch).toBeCalledWith('https://jsonplaceholder.typicode.com/posts', {
              method: 'DELETE',
              headers: { Authorization: `Bearer ${tokens.accessToken}`},
              body: JSON.stringify(params),
          });
  })


  it('auth fetch', async () => {
    const mockFetch = jest.fn().mockReturnValue({ok: true, json: () => {}});
    const saveToken = jest.fn().mockReturnValue({ json: () => { } });
    const getToken = jest.fn().mockReturnValue(Promise.resolve({accessToken: 'wvwvw', refreshToken: "dfbdfbfdfb"}));
    const http = new HttpService(mockFetch, getToken, saveToken);
    const email = "randomUser@mail.com"
    const password = "123123"

    await http.auth('https://jsonplaceholder.typicode.com/posts', email, password)
    expect(mockFetch).toBeCalledWith('https://jsonplaceholder.typicode.com/posts', {
              method: 'POST',
              headers: {'Accept': 'application/json', 'Content-Type': 'application/json'},
              body: JSON.stringify({email, password}),
          });
  })

  it('refreshTokens fetch', async () => {
    const mockFetch = jest.fn().mockReturnValue({ok: true, json: () => {}});
    const saveToken = jest.fn().mockReturnValue({ json: () => { } });
    const getToken = jest.fn().mockReturnValue(Promise.resolve({accessToken: 'wvwvw', refreshToken: "dfbdfbfdfb"}));
    const http = new HttpService(mockFetch, getToken, saveToken);
    const token = "nkgwjnrgwkj4234"

    await http.refreshToken(token)
    expect(mockFetch).toBeCalledWith('api/auth/refreshToken', {
              method: 'POST',
              headers: {'Accept': 'application/json', 'Content-Type': 'application/json'},
              body: JSON.stringify({token}),
          });
  })
  

    it('authGet error fetch', () => {
      const mockFetch = jest.fn().mockReturnValue({ok: false})
      const saveToken = jest.fn().mockReturnValue({  })
      const getToken = jest.fn().mockReturnValue(Promise.resolve({accessToken: 'wvwvw', refreshToken: "dfbdfbfdfb"}))
      const http = new HttpService(mockFetch, getToken, saveToken)

     http.authGet('wwwwwwww').catch(e => expect(e).toEqual(new ErrorFetch('authGet error',  404, true)))
    })


    it('authGet error 401', () => {
      const mockFetch =  jest.fn().mockReturnValueOnce({ok: false, status: 401}).mockReturnValueOnce({
        ok: true,
        json: () => {
          return {accessToken: 'wvwvw', refreshToken: "dfbdfbfdfb"};   
        }
      }).mockReturnValueOnce({ok: false, status: 401})
      const saveToken = jest.fn().mockReturnValue({  })
      const getToken = jest.fn().mockReturnValue(Promise.resolve({accessToken: 'wvwvw', refreshToken: "dfbdfbfdfb"}))
      const http = new HttpService(mockFetch, getToken, saveToken)

      http.authGet('wwwwwww').catch(e => expect(e).toEqual(new ErrorFetch('authGet error', 401, false)))
     })

    it('authPost error fetch', () => {
      const mockFetch = jest.fn().mockReturnValue({ok: false})
      const saveToken = jest.fn().mockReturnValue({  })
      const getToken = jest.fn().mockReturnValue(Promise.resolve({accessToken: 'wvwvw', refreshToken: "dfbdfbfdfb"}))
      const http = new HttpService(mockFetch, getToken, saveToken)
      const params = { 'Accept': 'application/json', 'Content-Type': 'application/json' }

     http.authPost('wwwwwwww', params).catch(e => expect(e).toEqual(new ErrorFetch('authPost error',  404, true)))
    })


    it('authPost error 401', () => {
      const mockFetch =  jest.fn().mockReturnValueOnce({ok: false, status: 401}).mockReturnValueOnce({
        ok: true,
        json: () => {
          return {accessToken: 'wvwvw', refreshToken: "dfbdfbfdfb"};   
        }
      }).mockReturnValueOnce({ok: false, status: 401})
      const saveToken = jest.fn().mockReturnValue({  })
      const getToken = jest.fn().mockReturnValue(Promise.resolve({accessToken: 'wvwvw', refreshToken: "dfbdfbfdfb"}))
      const http = new HttpService(mockFetch, getToken, saveToken)
      const params = { 'Accept': 'application/json', 'Content-Type': 'application/json' }

      http.authPost('wwwwwww', params).catch(e => expect(e).toEqual(new ErrorFetch('authPost error', 401, false)))
     })

    it('authPut error fetch', () => {
      const mockFetch = jest.fn().mockReturnValue({ok: false})
      const saveToken = jest.fn().mockReturnValue({  })
      const getToken = jest.fn().mockReturnValue(Promise.resolve({accessToken: 'wvwvw', refreshToken: "dfbdfbfdfb"}))
      const http = new HttpService(mockFetch, getToken, saveToken)
      const params = { 'Accept': 'application/json', 'Content-Type': 'application/json' }

     http.authPut('wwwwwwww', params).catch(e => expect(e).toEqual(new ErrorFetch('authPut error',  404, true)))
    })


    it('authPut error 401', () => {
      const mockFetch =  jest.fn().mockReturnValueOnce({ok: false, status: 401}).mockReturnValueOnce({
        ok: true,
        json: () => {
          return {accessToken: 'wvwvw', refreshToken: "dfbdfbfdfb"};   
        }
      }).mockReturnValueOnce({ok: false, status: 401})
      const saveToken = jest.fn().mockReturnValue({  })
      const getToken = jest.fn().mockReturnValue(Promise.resolve({accessToken: 'wvwvw', refreshToken: "dfbdfbfdfb"}))
      const http = new HttpService(mockFetch, getToken, saveToken)
      const params = { 'Accept': 'application/json', 'Content-Type': 'application/json' }

      http.authPut('wwwwwww', params).catch(e => expect(e).toEqual(new ErrorFetch('authPut error', 401, false)))
     })

    it('authDelete error fetch', () => {
      const mockFetch = jest.fn().mockReturnValue({ok: false})
      const saveToken = jest.fn().mockReturnValue({  })
      const getToken = jest.fn().mockReturnValue(Promise.resolve({accessToken: 'wvwvw', refreshToken: "dfbdfbfdfb"}))
      const http = new HttpService(mockFetch, getToken, saveToken)
      const params = { 'Accept': 'application/json', 'Content-Type': 'application/json' }

     http.authDelete('wwwwwwww', params).catch(e => expect(e).toEqual(new ErrorFetch('authDelete error',  404, true)))
    })


    it('authDelete error 401', () => {
      const mockFetch =  jest.fn().mockReturnValueOnce({ok: false, status: 401}).mockReturnValueOnce({
        ok: true,
        json: () => {
          return {accessToken: 'wvwvw', refreshToken: "dfbdfbfdfb"};   
        }
      }).mockReturnValueOnce({ok: false, status: 401})
      const saveToken = jest.fn().mockReturnValue({  })
      const getToken = jest.fn().mockReturnValue(Promise.resolve({accessToken: 'wvwvw', refreshToken: "dfbdfbfdfb"}))
      const http = new HttpService(mockFetch, getToken, saveToken)
      const params = { 'Accept': 'application/json', 'Content-Type': 'application/json' }

      http.authDelete('wwwwwww', params).catch(e => expect(e).toEqual(new ErrorFetch('authDelete error', 401, false)))
     })

    it('auth error fetch', () => {
      const mockFetch = jest.fn().mockReturnValue({ok: false})
      const saveToken = jest.fn().mockReturnValue({  })
      const getToken = jest.fn().mockReturnValue(Promise.resolve({accessToken: 'wvwvw', refreshToken: "dfbdfbfdfb"}))
      const http = new HttpService(mockFetch, getToken, saveToken)
      const email = "randomUser@mail.com"
      const password = "123123"

     http.auth('wwwwwwww', email, password).catch(e => expect(e).toEqual(new ErrorFetch('auth error',  404, true)))
    })


    it('auth error 401', () => {
      const mockFetch =  jest.fn().mockReturnValueOnce({ok: false, status: 401}).mockReturnValueOnce({
        ok: true,
        json: () => {
          return {accessToken: 'wvwvw', refreshToken: "dfbdfbfdfb"};   
        }
      }).mockReturnValueOnce({ok: false, status: 401})
      const saveToken = jest.fn().mockReturnValue({  })
      const getToken = jest.fn().mockReturnValue(Promise.resolve({accessToken: 'wvwvw', refreshToken: "dfbdfbfdfb"}))
      const http = new HttpService(mockFetch, getToken, saveToken)
      const email = "randomUser@mail.com"
      const password = "123123"

      http.auth('wwwwwww', email, password).catch(e => expect(e).toEqual(new ErrorFetch('auth error', 401, false)))
     })

     it('refreshToken error fetch', () => {
      const mockFetch = jest.fn().mockReturnValue({ok: false})
      const saveToken = jest.fn().mockReturnValue({  })
      const getToken = jest.fn().mockReturnValue(Promise.resolve({accessToken: 'wvwvw', refreshToken: "dfbdfbfdfb"}))
      const http = new HttpService(mockFetch, getToken, saveToken)
      const token = "nkgwjnrgwkj4234"

     http.refreshToken(token).catch(e => expect(e).toEqual(new ErrorFetch('refreshToken error',  404, true)))
    })


    it('refreshToken error 401', () => {
      const mockFetch =  jest.fn().mockReturnValueOnce({ok: false, status: 401}).mockReturnValueOnce({
        ok: true,
        json: () => {
          return {accessToken: 'wvwvw', refreshToken: "dfbdfbfdfb"};   
        }
      }).mockReturnValueOnce({ok: false, status: 401})
      const saveToken = jest.fn().mockReturnValue({  })
      const getToken = jest.fn().mockReturnValue(Promise.resolve({accessToken: 'wvwvw', refreshToken: "dfbdfbfdfb"}))
      const http = new HttpService(mockFetch, getToken, saveToken)
      const token = "nkgwjnrgwkj4234"

      http.refreshToken(token).catch(e => expect(e).toEqual(new ErrorFetch('refreshToken error', 401, false)))
     })


  describe('return value', () => {
    const mockFetch= jest.fn().mockReturnValue({
      ok: true,
      status: 200,
      json: () => {
        return {accessToken: 'wvwvw', refreshToken: "dfbdfbfdfb"};   
      }
    }
  );
     const saveToken = jest.fn().mockReturnValue({  })
     const getToken = jest.fn().mockReturnValue(Promise.resolve({accessToken: 'wvwvw', refreshToken: "dfbdfbfdfb"}))
     const http = new HttpService( mockFetch, getToken, saveToken)
     const params = { 'Accept': 'application/json', 'Content-Type': 'application/json' }
     const email = "randomUser@mail.com"
     const password = "123123"
     const token = "nkgwjnrgwkj4234"

     it('authGet', async ()=>{
      const data = await http.authGet('evwevw')
      expect(data).toEqual({accessToken: 'wvwvw', refreshToken: "dfbdfbfdfb"})
     })

     it('authPost', async ()=>{
      const data = await http.authPost('evwevw', params)
      expect(data).toEqual({accessToken: 'wvwvw', refreshToken: "dfbdfbfdfb"})
     })

     it('authPut', async ()=>{
      const data = await http.authPut('evwevw', params)
      expect(data).toEqual({accessToken: 'wvwvw', refreshToken: "dfbdfbfdfb"})
     })

     it('authDelete', async ()=>{
      const data = await http.authDelete('evwevw', params)
      expect(data).toEqual({accessToken: 'wvwvw', refreshToken: "dfbdfbfdfb"})
     })

     it('auth', async ()=>{
      const data = await http.auth('evwevw', email, password)
      expect(data).toEqual({accessToken: 'wvwvw', refreshToken: "dfbdfbfdfb"})
     })

     it('refreshToken', async ()=>{
      const data = await http.refreshToken(token)
      expect(data).toEqual({accessToken: 'wvwvw', refreshToken: "dfbdfbfdfb"})
     })
  })



 

