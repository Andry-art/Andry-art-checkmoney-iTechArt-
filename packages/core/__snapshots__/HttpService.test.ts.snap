// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`errors authGet 1`] = `"la la la"`;
